pkg/mons/restore_quorum.go

```diff
@@ -0,0 +1,291 @@
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if goodMonPublicIp == "" {
+		return fmt.Errorf("error: good mon %s not found", goodMon)
+	}
+
+	fsidSecret, err := context.Clientset.CoreV1().Secrets(clusterNamespace).Get(ctx.TODO(), "rook-ceph-mon", v1.GetOptions{})
+	if err != nil {
+		log.Fatalf("failed to get mon configmap %s %v", MonConfigMap, err)
+	}
+
+	cephFsid := string(fsidSecret.Data["fsid"])
+	if cephFsid == "" {
+		return fmt.Errorf("ceph cluster fsid not found")
+	}
+
+	fmt.Printf("printing fsid secret %s\n", cephFsid)
+	fmt.Println("Check for the running toolbox")
+
+	_, err = debug.GetDeployment(context, clusterNamespace, "rook-ceph-tools")
+	if err != nil {
+		return fmt.Errorf("failed to deployment rook-ceph-tools. %v", err)
+	}
+
+	toolBox, err := k8sutil.WaitForPodToRun(context, clusterNamespace, "app=rook-ceph-tools")
+	if err != nil || toolBox.Name == "" {
+		return fmt.Errorf("failed to get the running toolbox")
+	}
+
+	fmt.Printf("Restoring mon quorum to mon %s %s\n", goodMon, goodMonPublicIp)
+	fmt.Printf("The mons to discard are: %s\n", badMons)
+	fmt.Printf("The cluster fsid is %s\n", cephFsid)
+
+	var answer, output string
+	fmt.Printf("Are you sure you want to restore the quorum to mon %s? If so, enter 'yes-really-restore\n", goodMon)
+	fmt.Scanf("%s", &answer)
+	output, err = promptToContinueOrCancel(answer)
+	if err != nil {
+		return fmt.Errorf(" restoring the mon quorum to mon %s cancelled", goodMon)
+	}
+	fmt.Println(output)
+
+	fmt.Println("Waiting for operator pod to stop")
+	err = debug.SetDeploymentScale(context, operatorNamespace, "rook-ceph-operator", 0)
+	if err != nil {
+		return fmt.Errorf("failed to stop deployment rook-ceph-operator. %v", err)
+	}
+	fmt.Println("rook-ceph-operator deployment scaled down")
+
+	fmt.Println("Waiting for bad mon pod to stop")
+	for _, badMon := range badMons {
+		err = debug.SetDeploymentScale(context, clusterNamespace, fmt.Sprintf("rook-ceph-mon-%s", badMon), 0)
+		if err != nil {
+			return fmt.Errorf("deployment %s still exist. %v", fmt.Sprintf("rook-ceph-mon-%s", badMon), err)
+		}
+		fmt.Printf("deployment.apps/%s scaled\n", fmt.Sprintf("rook-ceph-mon-%s", badMon))
+	}
+
+	debug.StartDebug(context, clusterNamespace, fmt.Sprintf("rook-ceph-mon-%s", goodMon), "")
+
+	debugDeploymentSpec, err := debug.GetDeployment(context, clusterNamespace, fmt.Sprintf("rook-ceph-mon-%s-debug", goodMon))
+	if err != nil {
+		return fmt.Errorf("failed to deployment rook-ceph-mon-%s-debug", goodMon)
+	}
+
+	labelSelector := fmt.Sprintf("ceph_daemon_type=%s,ceph_daemon_id=%s", debugDeploymentSpec.Spec.Template.Labels["ceph_daemon_type"], debugDeploymentSpec.Spec.Template.Labels["ceph_daemon_id"])
+	_, err = k8sutil.WaitForPodToRun(context, clusterNamespace, labelSelector)
+	if err != nil {
+		return fmt.Errorf("failed to start deployment %s", fmt.Sprintf("rook-ceph-mon-%s-debug", goodMon))
+	}
+
+	fmt.Println("Started debug pod, restoring the mon quorum in the debug pod")
```

Comment:
How about factoring out lines 126-172 into a helper method for updating the monmap? That will help shorten this long method.

Reply:
refactored!